test/test_helper_discovery.rb

```diff
@@ -0,0 +1,26 @@
+def set_session_user_with_perms perms
+  perms = perms.collect{|p| FactoryGirl.create(:permission, :name => p) rescue Permission.find_by_name(p) }
```

Comment:
Does FactoryGirl fail when the permission already exist? so shouldn't we first try to find it instead of trying to create it?